File: base/selectionSOrt.js
function selectionSort(array) {
  const sortedList = [];
  const list = [...array]
  for(let i = 0; i < array.length; i++) {
    const index = indexOfMin(list);
    sortedList.push(list.splice(index,1)[0])
  }
  return sortedList;
}

function indexOfMin(items) {
  let minIndex = 0;

  for(let i = 0; i <  items.length; i++) {
    if (items[i] < items[minIndex]) {
      minIndex = i
    }
  }
  return minIndex;
}

const numbers = [12,23,3,2,55,6,44,21];
selectionSort(numbers)

File: teachers_note/selection_sort/selection_sort.java
import java.util.*;

public class Sort {
    static ArrayList<Integer> selectionSort(ArrayList<Integer> list) {
        ArrayList<Integer> sortedList = new ArrayList<Integer>();
        int length = list.size();
        for(int i = 0; i < length; i++) {
            int indexToMove = indexOfMin(list);
            int min = list.get(indexToMove);
            list.remove(indexToMove);
            sortedList.add(min);
        }
        return sortedList;
    }

    static int indexOfMin(ArrayList<Integer> list) {
        int minIndex = 0;
        int length = list.size();
        for (int i = 0; i < length; i++) {
            if(list.get(i) < list.get(minIndex)) {
                minIndex = i;
            }
        }
        return minIndex;
    }

    public static void main(String[] args) {
        ArrayList<Integer> list = new ArrayList<Integer>(Arrays.asList(29, 100, 1, 2, 57));
        System.out.println(selectionSort(list));
    }
}